corehq/motech/views.py

```diff
@@ -102,38 +111,163 @@ def page_url(self):
 
 
 @method_decorator(require_permission(Permissions.edit_motech), name='dispatch')
-class ConnectionSettingsView(BaseProjectSettingsView, FormView):
-    urlname = 'connection_settings_view'
-    page_title = ugettext_lazy('Connection Settings')
+class ConnectionSettingsListView(BaseProjectSettingsView, CRUDPaginatedViewMixin):
+    urlname = 'connection_settings_list_view'
+    page_title = _('Connection Settings')
     template_name = 'motech/connection_settings.html'
-    form_class = ConnectionSettingsFormSet  # NOTE: form_class is a formset
 
     def dispatch(self, request, *args, **kwargs):
+        # TODO: When Repeaters use Connection Settings, drop, and use
+        # @requires_privilege_with_fallback(privileges.DATA_FORWARDING)
         if not (
                 toggles.DHIS2_INTEGRATION.enabled_for_request(request)
                 or toggles.INCREMENTAL_EXPORTS.enabled_for_request(request)
         ):
             raise Http404()
-        return super(ConnectionSettingsView, self).dispatch(request, *args, **kwargs)
+        return super().dispatch(request, *args, **kwargs)
+
+    @property
+    def total(self):
+        return self.base_query.count()
+
+    @property
+    def base_query(self):
+        return ConnectionSettings.objects.filter(domain=self.domain)
+
+    @property
+    def column_names(self):
+        return [
+            _("Name"),
+            _("URL"),
+            _("Notify Addresses"),
+            _("Used By"),
+        ]
+
+    @property
+    def page_context(self):
+        return self.pagination_context
+
+    @property
+    def paginated_list(self):
+        for connection_settings in self.base_query.all():
+            yield {
+                "itemData": self._get_item_data(connection_settings),
+                "template": "connection-settings-template",
+            }
+
+    def _get_item_data(self, connection_settings):
+        return {
+            'id': connection_settings.id,
+            'name': connection_settings.name,
+            'url': connection_settings.url,
+            'notifyAddresses': ', '.join(connection_settings.notify_addresses),
+            'usedBy': ', '.join(connection_settings.used_by),
+            'editUrl': reverse(
+                ConnectionSettingsDetailView.urlname,
+                kwargs={'domain': self.domain, 'pk': connection_settings.id}
+            ),
+        }
+
+    def get_deleted_item_data(self, item_id):
+        connection_settings = ConnectionSettings.objects.get(
+            pk=item_id,
+            domain=self.domain,
+        )
+        if connection_settings.used_by:
+            raise Http409
+
+        connection_settings.delete()
```

Comment:
Is this supposed to be here?  I wouldn't expect `get_deleted_item_data` to also _do_ the deletion.

Reply:
It seems to be [how](https://github.com/dimagi/commcare-hq/blob/master/docs/ui_helpers.rst#allowing-deleting-in-your-paginated-view) [we](https://github.com/dimagi/commcare-hq/blob/32bfd28be8f5fd172a2c546d69e9e85a23c8e6a1/corehq/apps/settings/views.py#L579) [roll](https://github.com/dimagi/commcare-hq/blob/4766d4b5fb27f8584223b70257216b466ba5cada/corehq/apps/export/views/incremental.py#L88).